includes/menu.inc

```diff
@@ -5,25 +5,45 @@
  * Theme and preprocess functions for menus.
  */
 
+use Drupal\Core\Template\Attribute;
+
 /**
  * Implements hook_preprocess_menu().
  */
 function oe_bootstrap_theme_preprocess_menu(&$variables, $hook) {
   // No changes for menu toolbar.
-  if ($hook == 'menu__toolbar') {
+  if ($hook === 'menu__toolbar') {
     return;
   }
 
   // Get the current path.
   $current_path = \Drupal::request()->getRequestUri();
 
   $items = $variables['items'];
+
   foreach ($items as $key => $item) {
-    // If path is current_path, set active to li.
-    if ($item['url']->toString() == $current_path) {
-      // Add active link.
-      $variables['items'][$key]['attributes']['class'] = 'active';
+    $class = $item['in_active_trail'] ? ['active'] : [];
+    if (!empty($item['below'])) {
+      $class[] = 'nav-link';
+      $variables['items'][$key]['dropdown'] = TRUE;
+      $variables['items'][$key]['trigger']['label'] = $item['title'];
+      $variables['items'][$key]['trigger']['path'] = $item['url']->toString();
+      $variables['items'][$key]['trigger']['standalone'] = TRUE;
+      $variables['items'][$key]['trigger']['attributes'] = new Attribute(['class' => $class]);
+      $variables['items'][$key]['link'] = TRUE;
+      $variables['items'][$key]['alignment'] = 'dropdown-menu-sm-start';
+      foreach ($item['below'] as $key_subitem => $subitem) {
+        $subclass = $subitem['in_active_trail'] ? ['active'] : [];
+        $variables['items'][$key]['items'][$key_subitem]['label'] = $subitem['title'];
+        $variables['items'][$key]['items'][$key_subitem]['path'] = $subitem['url']->toString();
+        $variables['items'][$key]['items'][$key_subitem]['attributes'] = new Attribute(['class' => $subclass]);
+      }
+      continue;
     }
+    $variables['items'][$key]['label'] = $item['title'];
+    $variables['items'][$key]['path'] = $item['url']->toString();
+    // If path is current_path, set active to li.
```

Comment:
remove this comment, if the code is not clear then we try to improve code clarity and if its not possible we leave a comment, but here its clear that the menu item receives some classes (what does are is a different matter)